giraffe/src/utils/useBandHoverColumns.ts

```diff
@@ -0,0 +1,65 @@
+import {BandIndexMap, LineData, NumericColumnData} from '../types'
+import {useLazyMemo} from './useLazyMemo'
+import {isDefined} from './isDefined'
+
+const buildColumnData = (
+  lineData: LineData,
+  groupColData: NumericColumnData,
+  bandIndexMap: BandIndexMap
+) => {
+  const hoverableColumnData = {
+    xs: [],
+    ys: [],
+    groupColData: [],
+  }
+  const {rowIndices} = bandIndexMap
+
+  rowIndices.forEach(rowIndex => {
+    if (isDefined(rowIndex)) {
+      hoverableColumnData.xs = hoverableColumnData.xs.concat(
+        lineData[rowIndex].xs
+      )
```

Comment:
any reason not to go with the more slightly more compact?

```ts
hoverableColumnData.xs = [...hoverableColumnData.xs, ...lineData[rowIndex].xs)
```

Reply:
I like that. Thanks.

Reply:
Actually, i don't know if this will cause a problem: too many arguments passed to a function call.

This is literal array syntax, so not sure if the underlying implementation is a function call that creates a new array. If so, it will run into the "too many arguments" problem.

Reply:
Looks like it doesn't.

A literal array constructed from de-structuring other arrays uses iterable objects. So, we should be fine going with this suggestion.